docs/cli.md

```diff
@@ -0,0 +1,135 @@
+
+> [!NOTE]
+> The Model Registry stores the metadata about the models and images, but it does not store the actual model artifacts.
+> The model artifacts are stored in a separate storage location that is accessible to the ODH Edge platform such as an
+> S3 bucket or a Git repository.
+
+
+## Building ODH Edge CLI
+
+To build the ODH Edge CLI, run the following commands from the root of the repository: 
+
+```bash
+cd cli
+make cli-build
+```
+
+This will create a binary called `odh` in your current directory. You can move this binary to a location in your PATH to make it easier to run the ODH Edge CLI from anywhere.
+
+## Environment Setup
+
+### Prerequisites
+
+- Kubectl access to the OpenShift core cluster.
+- Go (version 1.20 or 1.21, version 1.22 is not supported).
+- [Model registry operator](https://github.com/opendatahub-io/model-registry-operator/releases/tag/v0.1.2) version 0.1.2 locally.
+
+### Installing Model Registry
+
+To install the Model Registry, follow the following steps:
+
+1. Change to the `model-registry-operator-0.1.2` directory:
+
+    ```bash
+    cd model-registry-operator-0.1.2
+    ```
+2. Install the CRDs into the cluster:
+
+    ```bash
+    make install
+    ```
+3. Deploy the Model Registry operator:
+
+    ```bash
+    make deploy
+    ```
+4. Create a Model Registry namespace:
+
+    ```bash
+   kubectl create ns odh-model-registry
+    ```
+5. Edit the `config/samples/mysql/modelregistry_v1alpha1_modelregistry.yaml` and change the `spec.rest.serviceRoute` to `enabled`.
+
+6. Create a Model Registry instance:
+
+    ```bash
+   kubectl -n odh-model-registry apply -k config/samples/mysql
+    ```
+
+7. Wait for the Model Registry pods to be ready:
+
+    ```bash
+    kubectl -n odh-model-registry get pods -w
+    ```
+
+8. Create a model-registry route:
+
+    ```bash
+    oc -n odh-model-registry expose service modelregistry-sample --port http-api
+    ```
+9. Get the route:
+
+    ```bash
+    oc -n odh-model-registry get route
+    ```
+
+## Usage
+
+The following flags are available globally for all commands:
+
+- `-m, --model-registry-url`: The URL of the Model Registry. This is the URL of the Model Registry service that is
+  running in the OpenShift cluster.
+- `k, --kubeconfig` : The path to the kubeconfig file. This is the kubeconfig file that is used to access the OpenShift 
+  cluster.
+
+### Listing Models
+
+To list all the models that are available in the Model Registry, run the following command:
+
+```bash
+odh -m <model-registry-url> models
+```
+
+### Adding a Model
+
+Adding a model to the model registry means adding the metadata about the model to the Model Registry. This metadata includes the model name, a model version and the pipeline parameters that are required to build the model image.
```

Comment:
Should we also mention the third part, which is the model artifact? By default the CLI will create an artifact with the same name as the model.

Reply:
I intentionally skipped documenting how we're using the model registry in this document and focused on the usage of the CLI from the user perspective. For the user, he provides a model name, the version, and the pipeline run parameters regardless of how they're stored in the model registry. The artifact concept isn't exposed to the CLI users.